cmd/thanos/query.go

```diff
@@ -391,12 +391,12 @@ func runQuery(
 		ctx, cancel := context.WithCancel(context.Background())
 		g.Add(func() error {
 			return runutil.Repeat(dnsSDInterval, ctx.Done(), func() error {
-				ctx, cancel = context.WithTimeout(ctx, dnsSDInterval)
```

Comment:
Shadowing variables... the worst... :face_with_head_bandage: 

The fix would be just to add `:` (:  but happy with explicit version as well :+1: